### PR TITLE
ASYNC-IMPORT-61: Create Import Start REST endpoint

### PR DESCRIPTION
**Pull request by Stepa4man**
### Description (*)
Create web endpoint for starting import

### Fixed Issues
1. magento/magento2#61: Create Import Start REST endpoint

### Manual testing scenarios (*)
...will be filled later

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by nuzil** [comment]
Hi @Stepa4man 
Can you please sync branch with mainline?

File: app/code/Magento/ImportService/Api/Data/ImportConfigInterface.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    const BEHAVIOUR = 'behaviour';
+    const IMPORT_IMAGE_ARCHIVE = 'import_image_archive';
+    const IMPORT_IMAGES_FILE_DIR = 'import_images_file_dir';
+    const ALLOWED_ERROR_COUNT = 'allowed_error_count';
+    const VALIDATION_STRATEGY = 'validation_strategy';
+    const EMPTY_ATTRIBUTE_VALUE_CONSTANT = 'empty_attribute_value_constant';
+    const CSV_SEPARATOR = 'csv_separator';
+    const CSV_ENCLOSURE = 'csv_enclosure';
+    const CSV_DELIMITER = 'csv_delimiter';
+    const MULTIPLE_VALUE_SEPARATOR = 'multiple_value_separator';
```
Comment (nuzil) [inline app/code/Magento/ImportService/Api/Data/ImportConfigInterface.php]: Can you please align params list? 
As we are doing it dynamic, lets remove not dynamic attributes for now, so output will be:

{
    "importConfig": {
        "profile_uuid": "uuid",
        "behaviour": "add_update, delete, update, add, replace",
        "validation_strategy": "string",
        "allowed_error_count": 0,
"extension_attributes": {
}
}

File: app/code/Magento/ImportService/Model/ImportConfig.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        $this->setData(self::VALIDATION_STRATEGY, $validationStrategy);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getExtensionAttributes(): ImportConfigExtensionInterface
+    {
+        // TODO: Implement getExtensionAttributes() method.
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/ImportConfig.php]: Pls set here return $this->_getExtensionAttributes();

File: app/code/Magento/ImportService/Model/ImportConfig.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setExtensionAttributes(ImportConfigExtensionInterface $extension): void
+    {
+        // TODO: Implement setExtensionAttributes() method.
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/ImportConfig.php]: pls set return $this->_setExtensionAttributes($extensionAttributes);

**Comment by nuzil** [comment]
Hi @Stepa4man 
Sorru for delay, here is a final JSON for start import operation:

POST /V1/import/type/{type}/start/{uuid}

{
    "importConfig": {
        "behaviour": "add_update, delete, update, add, replace",
        "validation_strategy": "string",
        "allowed_error_count": 0,
        "import_image_archive": "string",
        "import_images_file_dir": "string"
    }
}

**Comment by nuzil** [comment]
https://github.com/magento/architecture/blob/d189590eb6d5c18a749676b8ae3c8fe9e6fc8525/design-documents/asynchronous-import/base-extension.md#main-endpoint

**Comment by Stepa4man** [comment]
> Hi @Stepa4man
> Sorru for delay, here is a final JSON for start import operation:
> 
> POST /V1/import/type/{type}/start/{uuid}
> 
> {
> "importConfig": {
> "behaviour": "add_update, delete, update, add, replace",
> "validation_strategy": "string",
> "allowed_error_count": 0,
> "import_image_archive": "string",
> "import_images_file_dir": "string"
> }
> }

Done

**Comment by nuzil** [comment]
@Stepa4man Can you make a back merge? THere are a conflicts with mainline

**Comment by Stepa4man** [comment]
> @Stepa4man Can you make a back merge? THere are a conflicts with mainline

@nuzil Done

**Pull request merged**